backends/ubpf/ubpfType.h

```diff
@@ -53,13 +53,15 @@ namespace UBPF {
         cstring getAsString();
 
         void declare(EBPF::CodeBuilder *builder, cstring id, bool asPointer) override;
+        void declareInit(EBPF::CodeBuilder *builder, cstring id, bool asPointer) override;
```

Comment:
it does not look like any of this code is needed, it is just the same as in the base class.
so I expect no changes to ubpfType.h are needed.

Reply:
> it does not look like any of this code is needed, it is just the same as in the base class.
> so I expect no changes to ubpfType.h are needed.

It is needed or p4c does not even compile.  If you back out the changes to ubpfType.h and .cpp, p4c fails to compile as shown below.

```c
[ 67%] Built target p4c-graphs
[ 72%] Built target gtestp4c
[ 98%] Built target p4c-dpdk
In file included from /home/hemant/pr/p4c/build/backends/ubpf/unified_p4c_ubpf_sources_1.cpp:8:0:
/home/hemant/pr/p4c/backends/ubpf/ubpfType.cpp: In member function  virtual EBPF::EBPFType* UBPF::UBPFTypeFactory::create(const IR::Type*) :
/home/hemant/pr/p4c/backends/ubpf/ubpfType.cpp:49:42: error: invalid new-expression of abstract class type  UBPF::UBPFListType 
             result = new UBPFListType(tpl);
                                          ^
In file included from /home/hemant/pr/p4c/backends/ubpf/ubpfBackend.cpp:25:0,
                 from /home/hemant/pr/p4c/build/backends/ubpf/unified_p4c_ubpf_sources_1.cpp:3:
/home/hemant/pr/p4c/backends/ubpf/ubpfType.h:74:11: note:   because the following virtual functions are pure within  UBPF::UBPFListType :
     class UBPFListType : public EBPF::EBPFType, public EBPF::IHasWidth {
           ^~~~~~~~~~~~
In file included from /home/hemant/pr/p4c/backends/ubpf/ubpfType.h:20:0,
                 from /home/hemant/pr/p4c/backends/ubpf/ubpfBackend.cpp:25,
                 from /home/hemant/pr/p4c/build/backends/ubpf/unified_p4c_ubpf_sources_1.cpp:3:
/home/hemant/pr/p4c/backends/ebpf/ebpfType.h:35:18: note: 	virtual void EBPF::EBPFType::declareInit(EBPF::CodeBuilder*, cstring, bool)
     virtual void declareInit(CodeBuilder* builder, cstring id, bool asPointer) = 0;
                  ^~~~~~~~~~~
backends/ubpf/CMakeFiles/p4c-ubpf.dir/build.make:62: recipe for target 'backends/ubpf/CMakeFiles/p4c-ubpf.dir/unified_p4c_ubpf_sources_1.cpp.o' failed
make[2]: *** [backends/ubpf/CMakeFiles/p4c-ubpf.dir/unified_p4c_ubpf_sources_1.cpp.o] Error 1
CMakeFiles/Makefile2:1817: recipe for target 'backends/ubpf/CMakeFiles/p4c-ubpf.dir/all' failed
make[1]: *** [backends/ubpf/CMakeFiles/p4c-ubpf.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```